src/c2goto/library/pthread_lib.c

```diff
@@ -277,7 +277,7 @@ __ESBMC_HIDE:;
   if(retval != NULL)
     *retval = __ESBMC_pthread_end_values[(int)thread];
 
-  __ESBMC_really_atomic_end();
+  __ESBMC_atomic_end();
```

Comment:
Why?

Reply:
Both  `__ESBMC_really_atomic_end()` and `__ESBMC_atomic_end()` have exactly the same semantics. However, while the first one is treated as an intrisinc function, the second one is treated as a GOTO instruction.

I have replaced ` __ESBMC_really_atomic_end();` with ` __ESBMC_atomic_end();` to keep consistency in the code since the function starts with  `__ESBMC_begin_end();`.